src/main/java/com/vmware/vipclient/i18n/base/instances/LocaleMessage.java

```diff
@@ -71,12 +85,68 @@ public LocaleMessage() {
      *            a language tag determines the display name's localization
      * @return a list contains the display names
      */
-    public List<String> getSupportedDisplayNameList(String displayLanguage) {
-        Map<String, String> dispNameMap = new LocaleService().getDisplayNamesFromCLDR(displayLanguage, 
-        		VIPCfg.getInstance().getMsgOriginsQueue().listIterator());
+    public List<String> getSupportedDisplayNameList(String displayLocale) {
+        Map<String, String> dispNameMap = getDisplayNamesFromCLDR(displayLocale);
         Collection<String> valueCollection = dispNameMap.values();
         List<String> dispNameList = new ArrayList<String>(valueCollection);
         return dispNameList;
     }
 
+    private Map<String, String> getRegionsFromCLDR(String locale){
+        Map<String, String> regionMap = null;
+        LocaleService localeService = new LocaleService(null);
+        regionMap = localeService.getRegionsFromCLDR(locale);
+        if (regionMap != null) {
+            return regionMap;
+        }
+        if (!LocaleUtility.isDefaultLocale(locale)) {
+            logger.info("Can't find regions for locale [{}], look for English regions as fallback!", locale);
+            regionMap = localeService.getRegionsFromCLDR(LocaleUtility.getDefaultLocale().toLanguageTag());
+            if (regionMap != null) {
+                new FormattingCacheService().addRegions(locale, regionMap);
+                logger.debug("Default locale's regions is cached for locale [{}]!\n\n", locale);
+            }
+        }
+        return regionMap;
+    }
```

Comment:
resolved:
Move the locale fallback and caching logic in the service layer

not resolved:
Use LocaleDTO to hold the display locale. Do not pass it as a parameter here.

As we discussed offline,  add a new class LocaleDTO in case parameter increase,  but won’t put locale into it.
Because 'locale' is a frequently changed parameter, put it into DTO will need new service instance multiple times when its value changed, and new instance is time consuming.